src/test/regress/sql/multi_shard_modify.sql

```diff
@@ -0,0 +1,153 @@
+--
+-- MULTI_SHARD_MODIFY
+--
+
+ALTER SEQUENCE pg_catalog.pg_dist_shardid_seq RESTART 350000;
+
+-- Create a new hash partitioned multi_shard_modify_test table and stage data into it.
+CREATE TABLE multi_shard_modify_test (
+        t_key integer not null,
+        t_name varchar(25) not null,
+        t_value integer not null);
+SELECT master_create_distributed_table('multi_shard_modify_test', 't_key', 'hash');
+SELECT master_create_worker_shards('multi_shard_modify_test', 4, 2);
+
+COPY multi_shard_modify_test (t_key, t_name, t_value) FROM STDIN WITH (FORMAT 'csv');
+1,san francisco,99
+2,istanbul,34
+3,paris,46
+4,london,91
+5,toronto,98
+6,london,44
+7,stockholm,21
+8,tallinn,33
+9,helsinki,21
+10,ankara,6
+11,karabuk,78
+12,kastamonu,37
+13,samsun,55
+14,rome,13
+15,madrid,1
+16,barcelona,8
+17,poznan,12
+31,kabul,4
+32,dhaka,62
+33,iamey,121
+34,muscat,77
+41,uppsala,-1
+42,malmo,-2
+101,tokyo,106
+102,new delhi,978
+201,taipei,556
+202,beijing,754
+\.
+
+-- Testing master_modify_multiple_shards
+-- Verify that master_modify_multiple_shards cannot be called in a transaction block
+BEGIN;
+SELECT master_modify_multiple_shards('DELETE FROM multi_shard_modify_test WHERE t_key > 10 AND t_key <= 13');
+ROLLBACK;
+
+-- Check that master_modify_multiple_shards cannot be called with non-distributed tables
+CREATE TEMPORARY TABLE temporary_nondistributed_table (col_1 integer,col_2 text);
+INSERT INTO temporary_nondistributed_table VALUES (37, 'eren'), (31, 'onder');
+SELECT master_modify_multiple_shards('DELETE FROM temporary_nondistributed_table WHERE col_1 = 37');
+
+-- commands with volatile functions in their quals
+SELECT master_modify_multiple_shards('DELETE FROM multi_shard_modify_test WHERE t_key = (random() * 1000)');
+SELECT master_modify_multiple_shards('DELETE FROM multi_shard_modify_test WHERE t_value = (random() * 1000)');
+
+-- commands with stable functions in their quals
+CREATE FUNCTION temp_stable_func() RETURNS integer AS 'SELECT 10;' LANGUAGE SQL STABLE;
```

Comment:
I'm a little late here, but `CREATE FUNCTION` [isn't propagated](https://github.com/citusdata/citus/issues/376) to workers.